include/aikido/planner/Problem.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef AIKIDO_PLANNER_PROBLEM_HPP_
+#define AIKIDO_PLANNER_PROBLEM_HPP_
+
+#include <string>
+#include "aikido/constraint/Testable.hpp"
+#include "aikido/statespace/StateSpace.hpp"
+
+namespace aikido {
+namespace planner {
+
+/// Base class for various planning problems.
+class Problem
+{
+public:
+  /// Constructor.
+  ///
+  /// \param[in] stateSpace State space that this problem associated with.
+  /// \param[in] constraint Trajectory-wide constraint that must be satisfied.
+  Problem(
+      statespace::ConstStateSpacePtr stateSpace,
+      constraint::ConstTestablePtr constraint = nullptr);
+
+  /// Destructor.
+  virtual ~Problem() = default;
+
+  /// Returns type of this planning problem.
+  virtual const std::string& getType() const = 0;
+
+  /// Returns const state space.
+  statespace::ConstStateSpacePtr getStateSpace() const;
+
+  /// Returns the constraint that must be satisfied throughout the trajectory.
+  constraint::ConstTestablePtr getConstraint() const;
```

Comment:
If the problem was created with nullptr, does it return a `SatisfiedPtr` or a nullptr?

Reply:
It returns `SatisfiedPtr`. Added note about this to the constructor.